src/installation/guides/chroot.md

```diff
@@ -0,0 +1,235 @@
+[mke2fs(8)](https://man.voidlinux.org/mke2fs.8),
+[mkfs.xfs(8)](https://man.voidlinux.org/mkfs.xfs.8),
+[mkfs.btrfs(8)](https://man.voidlinux.org/mkfs.btrfs.8) or whatever tools are
+necessary for your filesystem(s) of choice.
+
+[cfdisk(8)](https://man.voidlinux.org/cfdisk.8) and
+[fdisk(8)](https://man.voidlinux.org/fdisk.8) are available on live
+distributions for partitioning, but you may wish to use
+[gdisk(8)](https://man.voidlinux.org/gdisk.8) (from the package `gptfdisk`) or
+[parted(8)](https://man.voidlinux.org/parted.8) instead.
+
+> Note: For a UEFI booting system, make sure to create a UEFI System Partition
+> (ESP). The ESP should have the partition type "EFI System" (code `ef00`) and
+> be formatted as fat32 using
+> [mkfs.vfat(8)](https://man.voidlinux.org/mkfs.vfat.8).
+
+If unsure what partitions to create, create (1) 1GB in size with type `EFI
+System` (code EF00) (2) remainder of the drive with type `Linux Filesystem`
+(code 8300).
+
+Then format these partitions with fat32 and ext4 respectively.
+
+```
+# mkfs.vfat /dev/sda1
+# mkfs.ext4 /dev/sda2
+```
+
+### Create a New Root and Mount Filesystems
+
+This guide will assume the new root filesystem is mounted on `/mnt`. You may
+wish to mount it elsewhere.
+
+If using UEFI, mount the EFI System Partition as `/mnt/boot/efi`.
+
+For example, if `/dev/sda2` is to be mounted as `/` and `dev/sda1` is the EFI
+System Partition
+
+```
+# mount /dev/sda2 /mnt/
+# mkdir -p /mnt/boot/efi/
+# mount /dev/sda1 /mnt/boot/efi/
+```
+
+Initialize swap space, if desired using
+[mkswap(8)](https://man.voidlinux.org/mkswap.8).
+
+## Base Installation
+
+[Select a mirror to download packages
+from.](../../xbps/repositories/mirror/index.md) For simplicity, save this URL to
+a shell variable for later use. Append `/current` to this repository url to
+install a glibc distribution, append `/current/musl` for a musl distribution.
+
+> Note: musl binaries are not available for i686.
+
+For example, this is the url for glibc from the Tier 1 server in Finland,
+
+```
+# REPO=https://alpha.de.repo.voidlinux.org/current
+```
+
+xbps also needs to know what architecture is being installed, available options
+are `x86_64`, `x86_64-musl` and `i686` for PC architecture computers.
+
+For example,
+
+```
+# ARCH=x86_64
+```
+
+This architecture must be compatible with your current operating system, but
+does not need to be the same. If your host is running an x86_64 operating
+system, any of the three architectures can be installed (whether the host is
+musl or glibc), but an i686 host can only install i686 distributions.
+
+Use xbps to boot strap the distribution by installing the `base-system`
+metapackage.
+
+```
+# XBPS_ARCH=$ARCH xbps-install -S -r /mnt -R "$REPO" base-system
+```
+
+## Configuration
+
+### Entering the Chroot
+
+Mount the pseudo-filesystems needed for a chroot,
+
+```
+# mount -t proc /proc /mnt/proc
+# mount -t sysfs /sys /mnt/sys
```

Comment:
@ahesford: Hmm, this is different from what @q66 [suggested](https://github.com/void-linux/void-docs/pull/163#discussion_r395958797) for the 'musl' section of the Handbook. Should the Handbook be changed?

Reply:
Especially for `/sys`, there can be sub mounts that one might want propagated to the chroot, like efivarfs. (Others, like cgroup mounts, are probably not critical, but I suppose it depends on what's being done in the chroot). The `--rbind` does this in one pass.

The `--make-rslave`, according to the manual for `mount`, seems to prevent changes that might be made to `/sys` in the chroot from propagating back to the host.

I'd like to hear from @q66 about the recommendation against bind-mounting `/sys` and whether `--make-rslave` addresses the concern.

Reply:
>  for /sys, there can be sub mounts that one might want propagated to the chroot, like efivarfs

This seems like a compelling reason, but I'm going to wait until I can do more research before I change anything.

> The --make-rslave, according to the manual for mount, seems to prevent changes that might be made to /sys in the chroot from propagating back to the host.

This sounds undesirable especially in the case of efivarfs. In the "basic" setup where you're using the same computer to install the distribution and run it, you're going to want efivarfs changes to propagate back to the host.

Reply:
I don't think `rslave` prevents the kind of propagation you're worried about. For one, `efivarfs` exposes UEFI nvram, so writing new variable, changing variable or deleting them in the chroot will still be written to nvram regardless of whether or not the "host" `efivarfs` sees the charges. (I have a suspicion that the host `efivarfs` directory will be updated to show the proper set of variables when listed regardless of this flag.)

Think about `rslave` in the context of `/dev`. You can create devices or remove them in a chroot and the changes won't be reflected in the original `/dev`. But you can still write or read anything you want from the devices in the chroot; `rbind` won't prevent it. The state of the underlying hardware will be changed.

This flag may be moot here, because `sysfs` manipulations are probably more about writing and reading files than creating or deleting them. A good test might be to `rbind` and `rslave` and, from the slave mount, try something like a PCIe device removal that will cause the `sysfs` hierarchy to change. Will the master mount reflect the changes hierarchy?

Reply:
As an aside, I can't strongly defend `rslave` for `sysfs` except to note that it seems to be common practice on the Internet when `rbind` mounting.

Reply:
as of my understanding, slaving and its alternatives are only about propagating/preventing mounts and they have nothing to do with the contents, but slave is actually good to protect the host... if this would be about an actual code, then it wouldnt matter as long as the code is safe, but as this is in the docs, anything can happen, and imho therefore it is the right thing to promote good and secure practices :)

Reply:
+ slave means that unmounting in the host will be propagated to the slave, and wont make issues if the slave would be forgotten to be tidied up

Reply:
Thanks for the clarification. It definitely seems like the right choice if an rbind is used.

Reply:
Bind mounting seems like the way to go, but I'd like to hear from @q66 on why this should not be done with /sys or /proc.

Also, this whole discussion has made me realize that a bind-mounted /dev, and then manually mounted /dev/pts inside of it..

Reply:
looks like we're going with rbind and make-rslave